pkg/operator2/workload/sync_openshift_oauth_apiserver.go

```diff
@@ -0,0 +1,289 @@
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehash"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/status"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+)
+
+// OAuthAPIServerWorkload is a struct that holds necessary data to install OAuthAPIServer
+type OAuthAPIServerWorkload struct {
+	operatorClient operatorconfigclient.AuthenticationsGetter
+	// countNodes a function to return count of nodes on which the workload will be installed
+	countNodes nodeCountFunc
+	// ensureAtMostOnePodPerNode a function that updates the deployment spec to prevent more than
+	// one pod of a given replicaset from landing on a node.
+	ensureAtMostOnePodPerNode ensureAtMostOnePodPerNodeFunc
+	targetNamespace           string
+	targetImagePullSpec       string
+	operatorImagePullSpec     string
+	kubeClient                kubernetes.Interface
+	eventRecorder             events.Recorder
+	versionRecorder           status.VersionGetter
+}
+
+// NewOAuthAPIServerWorkload creates new OAuthAPIServerWorkload struct
+func NewOAuthAPIServerWorkload(
+	operatorClient operatorconfigclient.AuthenticationsGetter,
+	countNodes nodeCountFunc,
+	ensureAtMostOnePodPerNode ensureAtMostOnePodPerNodeFunc,
+	targetNamespace string,
+	targetImagePullSpec string,
+	operatorImagePullSpec string,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+	versionRecorder status.VersionGetter,
+) *OAuthAPIServerWorkload {
+	return &OAuthAPIServerWorkload{
+		operatorClient:            operatorClient,
+		countNodes:                countNodes,
+		ensureAtMostOnePodPerNode: ensureAtMostOnePodPerNode,
+		targetNamespace:           targetNamespace,
+		targetImagePullSpec:       targetImagePullSpec,
+		operatorImagePullSpec:     operatorImagePullSpec,
+		kubeClient:                kubeClient,
+		eventRecorder:             eventRecorder,
+		versionRecorder:           versionRecorder,
+	}
+}
+
+// Sync essentially manages OAuthAPI server.
+func (c *OAuthAPIServerWorkload) Sync() (*appsv1.Deployment, []error) {
+	errs := []error{}
+
+	authOperator, err := c.operatorClient.Authentications().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	// TODO: block until config is obvserved when required
+	/*if operatorCfg, err := getStructuredConfig(authOperator.Spec.OperatorSpec); err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	} else {
+		if len(operatorCfg.APIServerArguments) == 0 {
+			klog.Info("Waiting for observed configuration to be available")
+			errs = append(errs, errors.New("waiting for observed configuration to be available (spec.ObservedConfig.Raw)"))
+			return nil, errs
+		}
+	}*/
```

Comment:
why commented?

Reply:
config support hasn't been wired yet, @stlaz is working on that https://github.com/openshift/library-go/pull/733